src/bindings/python/src/openvino/runtime/passes/manager.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from openvino.pyopenvino.passes import Manager as ManagerBase
+from openvino.pyopenvino.passes import PassBase
+
+
+class Manager(ManagerBase):
+    """Manager that additionally holds transformations objects."""
+    def __init__(self):
+        super().__init__()
+        self.passes_list = []  # need to keep python instances alive
+
+    def register_pass(self, *args, **kwargs):
+        for arg in args:
+            if isinstance(arg, PassBase):
+                self.passes_list.append(arg)
+
+        for arg in kwargs.values():
+            if isinstance(arg, PassBase):
+                self.passes_list.append(arg)
+
+        return super().register_pass(*args, **kwargs)
```

Comment:
I understand here the idea of args and kwargs comes from... there are many specialized overloads to it. However I think this should be created in more error-prone fashion, as now it will always add `PassBase` even if following arguments do not match this specific overload.

Example:
```python
m.register_pass("some stringy", MyPass())
```
The MyPass will still be added and not match any of the parent method. It is not critical when we use code that is failing in runtime, more problem arises with try/catch approaches or different threads.

This should be re-written in my opinion. Maybe to some if/else statement - depending on if first (and only) arg/kwarg is matching this specific overload (then add to list and trigger the register_pass already in the next line), otherwise it can trigger just parent method (without appending!).

Reply:
Hmm, I've added tests for negative cases like you described and everything works fine. Please take a look at the tests I've added: https://github.com/openvinotoolkit/openvino/pull/10971/files#diff-3f2632473a0ef407e0ce1f4f68538b652fd4f02f189c025caf27261b98458924R39-R44
So I think we can keep current solution with arg/kwarg as it is already works as expected.